core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1303,4 +1303,10 @@ package object config {
     .doc("Staging directory used while submitting applications.")
     .stringConf
     .createOptional
+
+  private[spark] val BUFFER_PAGESIZE = ConfigBuilder("spark.buffer.pageSize")
+    .doc("The amount of memory used per page in bytes")
+    .bytesConf(ByteUnit.BYTE)
```

Comment:
Nit: have you considered defining its unit as MiB? 

Like:
```
    .bytesConf(ByteUnit.MiB)
```

Reply:
@attilapiros 
At the destination(for allocation of pagesize), we need in Byte unit only.
And user can still configure with their values and unit. ex: 1k,1m,1g
hope addressed your query.

Reply:
I see, especially as this value is used as default for `memoryManager.pageSizeBytes` too.

Reply:
If someone specified a value like "1000" that would be interpreted differently if the default units were MiB

Reply:
Yes, as it is an already existing parameter. My bad.